integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/JdbcSparkUtils.java

```diff
@@ -40,11 +40,20 @@ public static <D extends OpenLineage.Dataset> List<D> getDatasets(
     Properties jdbcProperties = relation.jdbcOptions().asConnectionProperties();
 
     if (meta.columnLineage().isEmpty()) {
-      DatasetIdentifier di =
-          JdbcDatasetUtils.getDatasetIdentifier(
-              jdbcUrl, meta.inTables().get(0).qualifiedName(), jdbcProperties);
-      return Collections.singletonList(
-          datasetFactory.getDataset(di.getName(), di.getNamespace(), schema));
+      int numberOfTables = meta.inTables().size();
+
+      return meta.inTables().stream()
+          .map(
+              dbtm -> {
+                DatasetIdentifier di =
+                    JdbcDatasetUtils.getDatasetIdentifier(
+                        jdbcUrl, dbtm.qualifiedName(), jdbcProperties);
+                return datasetFactory.getDataset(
+                    di.getName(),
+                    di.getNamespace(),
+                    numberOfTables == 1 ? schema : new StructType());
```

Comment:
Won't this result in schema facet with empty fields? If so, this is something that should be avoided.
 
If any other OL event emits dataset with schema facet, the schema will be attached to a dataset up until the moment a new schema facet is sent. Sending empty schema facet will clear existing information.

Reply:
That example is only possible to happen when we have no column lineage + number of tables in the jdbc query is more than one. Maybe instead this solution we should take one step backward and add column level lineage for jdbc queries with with statement ? I don't see any better solution without stepping back than putting the empty Schema field. Do you have any ideas ?

Reply:
How about adding new method `io.openlineage.spark.api.DatasetFactory#getDataset(java.lang.String, java.lang.String)` without `StructType schema` parameter?

Reply:
fixed